### ISSUE TITLE
Bug with array_merge_recursive

### ISSUE THREAD

**Issue by weierophinney** [laminas/laminas-servicemanager#4]
I'm trying to do something like this:

```php
$config1 = [
    'factories`=> [
        // factories
    ],
    'lazy_services' => [
        'class_map' => [
            'Foo' => 'Foo',
        ],
    ],
];
```

```php
$config2 = [
    'factories`=> [
        // factories
    ],
    'lazy_services' => [
        'class_map' => [
            'Foo' => 'Foo',
        ],
    ],
];

$container = new ServiceManager($config1);
$container->configure($config2);
```

This will result on internal `lazy_services` property the following:

```php
$this->lazy_services = [
    'class_map' => [
        'Foo' => [
            'Foo',
            'Foo',
        ],
    ].
];
```

The problem is the use of `array_merge_recursive()` function in [this line](https://github.com/zendframework/zend-servicemanager/blob/a1ed6140d0d3ee803fec96582593ed024950067b/src/ServiceManager.php#L360).

Same problem with `delegators` property.

I found this issue using `zend-mvc` and an already configured container, then `Zend\ModuleManager\Listener\ServiceListener` reconfigure it adding configuration from ModuleManager features.

---

Originally posted by @thomasvargiu at https://github.com/zendframework/zend-servicemanager/issues/279

**Comment by GeeH** [comment]
Dupe of #3 fixed by #43

**Comment by settermjd** [comment]
I have a similar issue with delegators and have a simple patch for it. Just need help with testing the change.
